cfn-resources/test/e2e/utility/util.go

```diff
@@ -49,21 +56,43 @@ func PublishToPrivateRegistry(t *testing.T, rctx ResourceContext) {
 	t.Setenv("RESOURCE_TYPE_NAME_FOR_E2E", rctx.ResourceTypeNameForE2e)
 	t.Setenv("E2E_RAND_SUFFIX", rctx.E2eRandSuffix)
 	t.Setenv("RESOURCE_DIRECTORY_NAME", rctx.ResourceDirectory)
+	t.Cleanup(func() {
+		RunCleanupScript(t, rctx)
+	})
 
-	output, err := runShScript("../utility/publish_cfn_to_registry.sh")
+	output, err := runShScript(t, "../utility/publish_cfn_to_registry.sh")
 	FailNowIfError(t, fmt.Sprintf("Error when executing publishing script. Output: %s\n", output)+"%v", err)
-
+	if err != nil {
+		return
+	}
 	t.Logf("New E2E test resource type %s successfully published to private registry!\n", rctx.ResourceTypeNameForE2e)
+}
 
-	t.Cleanup(func() {
-		RunCleanupScript(t, rctx)
-	})
+func runShScript(t *testing.T, path string) ([]byte, error) {
+	cmd := exec.Command(path)
+	stdout, _ := cmd.StdoutPipe()
+	stderr, _ := cmd.StderrPipe()
+	err := cmd.Start()
+	if err != nil {
+		return nil, err
+	}
+
+	// Read the output from pipes
+	output, _ := readOutput(stdout)
+	errorOutput, _ := readOutput(stderr)
+
+	err = cmd.Wait()
```

Comment:
```suggestion
	cmd := exec.Command(path)
	resp, err := cmd.CombinedOutput()
```

Reply:
@gssbzn We had CombinedOutput before but that didn't seem to properly return errors or log anything from the script.
See https://github.com/mongodb/mongodbatlas-cloudformation-resources/blob/master/cfn-resources/test/e2e/utility/util.go#L64 Let me know if there's a better way

Reply:
the difference is you were capturing the output of `sh` not the binary, if the binary wasn't piping to sh you would get nothing, also since you are coming from that sh script, consider here whatever you are passing to `exec.Command` won't necessarily have access to the env variables and you may need to do `cmd.Env = os.Environ()`

Reply:
updated